### ISSUE TITLE
Q: Best way of adding a loss function to dlib that depends on the evaluation of another network

### ISSUE THREAD

**Issue by OranjeeGeneral** [davisking/dlib#492]
Hi 
I want to add a new loss function so I can use it in a neural network but the loss function depends on the evaluation of a second neural network what would be the best way to do it?

It looks like I would have to extend loss interface a bit as I can't see how I would be able to define a reference to a second network structure.

**Comment by davisking** [comment]
You can stick any objects you want into the loss object.  So if you want to stick 10 other network objects in there then you can do that.

**Comment by OranjeeGeneral** [comment]
So the best way would be to copy the mmod_loss and have an additional options support class?

**Comment by davisking** [comment]
I don't know.  It depends on what you want to do.  But you can literally put anything you want into a loss object.
